cmd/relayer/main.go

```diff
@@ -0,0 +1,412 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	rcontracts "github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	net "github.com/athanorlabs/go-relayer/net"
+	"github.com/athanorlabs/go-relayer/relayer"
+	rrpc "github.com/athanorlabs/go-relayer/rpc"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	swapnet "github.com/athanorlabs/atomic-swap/net"
+
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	flagDataDir          = "datadir"
+	flagEndpoint         = "endpoint"
```

Comment:
Is this the ETH endpoint?  We call this flag `ethereum-endpoint` in swapd, which is a little on the long side to type. Would `eth-endpoint` in both places be less confusing, but easy to type and consistent across programs?

Reply:
the relayer is eth-only, so I omitted the "ethereum" part for a few of the flags. do you think I should change the flags, or make it clearer what the relayer is for?

Reply:
There are 3 endpoints associated with the relayer, even if the flags for the other 2 don't use the phrase "endpoint", so I think having a prefix would be more clear:

1. Ethereum endpoint
2. RPC endpoint
3. libp2p endpoint